src/App.tsx

```diff
@@ -15,48 +17,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  sortFieldName = 'name',
+  sortFieldLength = 'length',
+  sortFieldNone = '',
+}
+
+type SortParams = {
+  sortField: SortField;
+  isReversed: boolean;
+};
+
+function sortGoodsByParams(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+): string[] {
+  const orderedField = [...goods];
+
+  if (sortField) {
+    orderedField.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.sortFieldName:
+          return good1.localeCompare(good2);
+
+        case SortField.sortFieldLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    orderedField.reverse();
+  }
+
+  return orderedField;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortField.sortFieldNone);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleOrder = sortGoodsByParams(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortField !== SortField.sortFieldName },
+          )}
+          onClick={() => setSortField(SortField.sortFieldName)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button is-success',
+            { 'is-light': sortField !== SortField.sortFieldLength },
+          )}
+          onClick={() => setSortField(SortField.sortFieldLength)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button is-warning',
+            { 'is-light': isReversed === false },
+          )}
+          onClick={() => {
+            setIsReversed(reversed => !reversed);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {((sortField !== '' || isReversed === true) && (
+          <button
+            type="button"
+            className={cn('button is-danger is-light')}
+            onClick={() => {
+              setSortField(SortField.sortFieldNone);
+              setIsReversed(false);
+            }}
```

Comment:
same for this arrow function
```suggestion
            onClick={onReset}
```